stdlib/public/core/StringStorage.swift

```diff
@@ -376,7 +436,12 @@ extension _StringStorage {
 
   @nonobjc
   internal func clear() {
+#if arch(i386) || arch(arm)
+    self._count = 0
+    self._flags = Flags(isASCII: true)
```

Comment:
Yeah, I'm thinking about refactoring much of this to take a `Flags` struct that append will `&` together, and `clear()` can set to be `Flags.emptyString` or something, which will be ASCII.